sabnzbd/downloader.py

```diff
@@ -372,12 +372,17 @@ def remove_socket(self, nw: NewsWrapper):
     @NzbQueueLocker
     def set_paused_state(self, state: bool):
         """Set downloader to specified paused state"""
-        self.paused = state
+        if self.paused != state:
```

Comment:
What happens here? Why is this needed?
This code needs come comments.

Reply:
It's not strictly needed but it avoids doing the check and update if there is no change. It's also not necessary to set paused if it's unchanged so I moved it to the same level.